unison-syntax/src/Unison/Parser/Ann.hs

```diff
@@ -20,9 +20,28 @@ instance Monoid Ann where
   mempty = External
 
 instance Semigroup Ann where
-  Ann s1 _ <> Ann _ e2 = Ann s1 e2
+  Ann s1 e1 <> Ann s2 e2 = Ann (min s1 s2) (max e1 e2)
```

Comment:
We really want the semigroup here to be Ann Union, since it's typically used to generate the span of all children of the current node, E.g. the existing semigroup resulted in invalid ranges for operations like `(ann op <> ann lhs <> ann rhs)` on an infix operator, and type arrows compute an annotation where the start is AFTER the end.